src/preprocessing/passes/bool_to_bv.cpp

```diff
@@ -39,165 +39,159 @@ PreprocessingPassResult BoolToBV::applyInternal(
+    newAssertion = nm->mkNode(kind::EQUAL, newAssertion, bv::utils::mkOne(1));
+    newAssertionType = newAssertion.getType();
+  }
+  Assert(newAssertionType.isBoolean());
+  return newAssertion;
+}
+
+Node BoolToBV::lowerNode(const TNode& node, bool force)
 {
-  bool optionITE = options::boolToBitvector() == BOOL_TO_BV_ITE;
-  NodeManager* nm = NodeManager::currentNM();
   std::vector<TNode> visit;
-  visit.push_back(a);
+  visit.push_back(node);
   std::unordered_set<TNode, TNodeHashFunction> visited;
-  // for ite mode, keeps track of whether you're in an ite condition
-  // for all mode, unused
-  std::unordered_set<TNode, TNodeHashFunction> ite_cond;
 
   while (!visit.empty())
   {
     TNode n = visit.back();
     visit.pop_back();
 
-    int numChildren = n.getNumChildren();
-    Kind k = n.getKind();
-    Debug("bool-to-bv") << "BoolToBV::lowerAssertion Post-traversal with " << n
-                        << " and visited = " << ContainsKey(visited, n)
+    Debug("bool-to-bv") << "BoolToBV::lowerNode: Post-order traversal with "
+                        << n << " and visited = " << ContainsKey(visited, n)
                         << std::endl;
 
+    int numChildren = n.getNumChildren();
+
     // Mark as visited
-    /* Optimization: if it's a leaf, don't need to wait to do the work */
-    if (!ContainsKey(visited, n) && (numChildren > 0))
+    if (!ContainsKey(visited, n))
     {
       visited.insert(n);
       visit.push_back(n);
 
       // insert children in reverse order so that they're processed in order
-      //     important for rewriting which sorts by node id
+      //    important for rewriting which sorts by node id
       for (int i = numChildren - 1; i >= 0; --i)
       {
         visit.push_back(n[i]);
       }
-
-      if (optionITE)
-      {
-        // check for ite-conditions
-        if (k == kind::ITE)
-        {
-          ite_cond.insert(n[0]);
-        }
-        else if (ContainsKey(ite_cond, n))
-        {
-          // being part of an ite condition is inherited from the parent
-          ite_cond.insert(n.begin(), n.end());
-        }
-      }
-    }
-    /* Optimization for ite mode */
-    else if (optionITE && !ContainsKey(ite_cond, n) && !needToRebuild(n))
-    {
-      Debug("bool-to-bv")
-          << "BoolToBV::lowerAssertion Skipping because don't need to rebuild: "
-          << n << std::endl;
-      // in ite mode, if you've already visited the node but it's not
-      // in an ite condition and doesn't need to be rebuilt, then
-      // don't need to do anything
-      continue;
     }
     else
     {
-      lowerNode(n);
+      lowerNodeHelper(n, force);
     }
   }
 
-  if (fromCache(a).getType().isBitVector())
-  {
-    return nm->mkNode(kind::EQUAL, fromCache(a), bv::utils::mkOne(1));
-  }
-  else
-  {
-    Assert(a == fromCache(a));
-    return a;
-  }
+  return fromCache(node);
 }
 
-void BoolToBV::lowerNode(const TNode& n)
+void BoolToBV::lowerNodeHelper(const TNode& n, bool force)
```

Comment:
This is short and only used once -- why not inline it?

Reply:
Note: renamed this to `visit` based on other comments. You're right that it's only used once, but it's actually pretty long (that first return is just one of the exit conditions).